pkg/services/sqlstore/dashboard_snapshot.go

```diff
@@ -45,22 +49,32 @@ func CreateDashboardSnapshot(cmd *models.CreateDashboardSnapshotCommand) error {
 			expires = time.Now().Add(time.Second * time.Duration(cmd.Expires))
 		}
 
-		snapshot := &models.DashboardSnapshot{
-			Name:              cmd.Name,
-			Key:               cmd.Key,
-			DeleteKey:         cmd.DeleteKey,
-			OrgId:             cmd.OrgId,
-			UserId:            cmd.UserId,
-			External:          cmd.External,
-			ExternalUrl:       cmd.ExternalUrl,
-			ExternalDeleteUrl: cmd.ExternalDeleteUrl,
-			Dashboard:         cmd.Dashboard,
-			Expires:           expires,
-			Created:           time.Now(),
-			Updated:           time.Now(),
+		encode, err := cmd.Dashboard.Encode()
+		if err != nil {
+			return err
 		}
 
-		_, err := sess.Insert(snapshot)
+		encryptedDashboard, err := util.Encrypt(encode, setting.SecretKey)
```

Comment:
Right now the logic for encrypt/decrypt is spread out between sqlstore and api package. I would suggest introducing a new package `securedata` and related struct for holding the encrypted data  and provide methods for encrypt and decrypt. This would be very similar to the `securejsondata` package. 

Example usage of the securejsondata can be seen for datasources: 
https://github.com/grafana/grafana/blob/90303b5a03215f5dbcd3215a182f3251e0f0a53c/pkg/services/sqlstore/datasource.go#L140
https://github.com/grafana/grafana/blob/90303b5a03215f5dbcd3215a182f3251e0f0a53c/pkg/models/datasource_cache.go#L251

WDYT?

Reply:
I like it 👍